docs/csharp/language-reference/keywords/readonly.md

```diff
@@ -9,50 +9,86 @@ helpviewer_keywords:
 ms.assetid: 2f8081f6-0de2-4903-898d-99696c48d2f4
 ---
 # readonly (C# Reference)
-The `readonly` keyword is a modifier that you can use on fields. When a field declaration includes a `readonly` modifier, assignments to the fields introduced by the declaration can only occur as part of the declaration or in a constructor in the same class.  
-  
+
+The `readonly` keyword is a modifier that can be used in three contexts:
+
+- In a [field declaration](#readonly-field-example), `readonly` indicates that assignment to the field can only occur as part of the declaration or in a constructor in the same class.
+- In a [`readonly struct` definition](#readonly-struct-example), `readonly` indicates that the `struct` is immutable.
+- In a [`ref readonly` method return](#ref-readonly-return-example), the `readonly` modifier indicates that method returns a reference and writes are not allowed to that reference.
+
+The final two contexts were added in C# 7.2.
+
 ## Readonly field example  
- In this example, the value of the field `year` cannot be changed in the method `ChangeYear`, even though it is assigned a value in the class constructor:  
-  
- [!code-csharp[csrefKeywordsModifiers#14](../../../csharp/language-reference/keywords/codesnippet/CSharp/readonly_1.cs)]  
-  
- You can assign a value to a `readonly` field only in the following contexts:  
+
+In this example, the value of the field `year` cannot be changed in the method `ChangeYear`, even though it is assigned a value in the class constructor:  
   
--   When the variable is initialized in the declaration, for example:  
+[!code-csharp[Readonly Field example](~/samples/snippets/csharp/keywords/ReadonlyKeywordExamples.cs#ReadonlyField)]  
   
-    ```csharp  
-    public readonly int y = 5;  
-    ```  
+You can assign a value to a `readonly` field only in the following contexts:  
   
--   For an instance field, in the instance constructors of the class that contains the field declaration, or for a static field, in the static constructor of the class that contains the field declaration. These are also the only contexts in which it is valid to pass a `readonly` field as an [out](../../../csharp/language-reference/keywords/out-parameter-modifier.md) or [ref](../../../csharp/language-reference/keywords/ref.md) parameter.  
+- When the variable is initialized in the declaration, for example:  
+
+```csharp
+public readonly int y = 5;  
+```
+
+- For an instance field, in the instance constructors of the class that contains the field declaration, or for a static field, in the static constructor of the class that contains the field declaration. These are also the only contexts in which it is valid to pass a `readonly` field as an [out](out-parameter-modifier.md) or [ref](ref.md) parameter.  
```

Comment:
This first sentence is kind of awkward. I'm not sure what you are trying to say.